From 0f9271140874c88bd15bfb634a1867de5d77d3fe Mon Sep 17 00:00:00 2001
From: Tanuj Tekriwal <tanuj.tekriwal@intel.com>
Date: Wed, 11 Sep 2024 08:15:31 +0000
Subject: [PATCH] WA Disable trusty for A15 BM boot

This patch will disable trusty for A15 BM boot

Tests Done: MTL BM boots fine

Tracked-On: NA
Signed-off-by: Tanuj Tekriwal <tanuj.tekriwal@intel.com>
---
 groups/device-specific/caas/BoardConfig.mk           |  2 +-
 groups/device-specific/caas/framework_manifest.xml   |  8 --------
 .../config-lts/lts2022-chromium/x86_64_defconfig     | 12 +-----------
 3 files changed, 2 insertions(+), 20 deletions(-)

diff --git a/groups/device-specific/caas/BoardConfig.mk b/groups/device-specific/caas/BoardConfig.mk
index f93ac00..5047504 100644
--- a/groups/device-specific/caas/BoardConfig.mk
+++ b/groups/device-specific/caas/BoardConfig.mk
@@ -52,4 +52,4 @@ BUILD_BROKEN_USES_BUILD_HOST_SHARED_LIBRARY := true
 BUILD_BROKEN_USES_BUILD_HOST_EXECUTABLE := true
 BUILD_BROKEN_USES_BUILD_COPY_HEADERS := true
 # PRODUCT_COPY_FILES directives.
-BUILD_BROKEN_ELF_PREBUILT_PRODUCT_COPY_FILES := true
+BUILD_BROKEN_ELF_PREBUILT_PRODUCT_COPY_FILES := false
diff --git a/groups/device-specific/caas/framework_manifest.xml b/groups/device-specific/caas/framework_manifest.xml
index 71ea596..bae0cbe 100644
--- a/groups/device-specific/caas/framework_manifest.xml
+++ b/groups/device-specific/caas/framework_manifest.xml
@@ -52,14 +52,6 @@
         <version>1</version>
         <fqname>IThermal/default</fqname>
     </hal>
-    <hal format="aidl">
-        <name>android.hardware.gatekeeper</name>
-        <version>1</version>
-        <interface>
-            <name>IGatekeeper</name>
-            <instance>default</instance>
-        </interface>
-    </hal>
     <hal format="aidl">
         <name>android.hardware.audio.core</name>
         <version>2</version>
diff --git a/groups/kernel/gmin64/config-lts/lts2022-chromium/x86_64_defconfig b/groups/kernel/gmin64/config-lts/lts2022-chromium/x86_64_defconfig
index 92554cc..22fbc6c 100644
--- a/groups/kernel/gmin64/config-lts/lts2022-chromium/x86_64_defconfig
+++ b/groups/kernel/gmin64/config-lts/lts2022-chromium/x86_64_defconfig
@@ -3470,17 +3470,7 @@ CONFIG_PROC_THERMAL_MMIO_RAPL=y
 #
 # Trusty driver
 #
-CONFIG_TRUSTY=y
-# CONFIG_TRUSTY_IRQ is not set
-CONFIG_TRUSTY_LOG=y
-# CONFIG_TRUSTY_TEST is not set
-CONFIG_TRUSTY_VIRTIO=y
-CONFIG_TRUSTY_VIRTIO_IPC=y
-# CONFIG_TRUSTY_DMA_BUF_FFA_TAG is not set
-# CONFIG_TRUSTY_DMA_BUF_SHARED_MEM_ID is not set
-# CONFIG_TRUSTY_CRASH_IS_PANIC is not set
-CONFIG_TRUSTY_X86_64=y
-CONFIG_TRUSTY_BACKUP_TIMER=y
+# CONFIG_TRUSTY is not set
 # end of Trusty driver
 
 CONFIG_WATCHDOG=y
-- 
2.34.1

